OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs

```diff
@@ -317,7 +331,11 @@ void RefreshServerList()
 						if (addressNode != null)
 							addressNode.Value.Value = bl.Address.ToString().Split(':')[0] + ":" + addressNode.Value.Value.Split(':')[1];
 
-						lanGames.Add(new GameServer(game));
+						try
+						{
+							lanGames.Add(new GameServer(game));
+						}
+						catch { }
```

Comment:
What is going on here? Why do we suddenly `try` `catch` this and discard any error?

Reply:
This fixes a crash/exploit where sending malformed game advertisements will kill any other clients on the LAN.  I split this fix into its own commit to make it clearer.